src/channel_type.rs

```diff
@@ -0,0 +1,201 @@
+use bytes::{Buf, BufMut};
+
+use crate::marshal::{Marshal, MarshalSize, Unmarshal};
+
+const CHANNEL_TYPE_RELIABLE: u8 = 0x00;
+const CHANNEL_TYPE_RELIABLE_UNORDERED: u8 = 0x80;
+const CHANNEL_TYPE_PARTIAL_RELIABLE_REXMIT: u8 = 0x01;
+const CHANNEL_TYPE_PARTIAL_RELIABLE_REXMIT_UNORDERED: u8 = 0x81;
+const CHANNEL_TYPE_PARTIAL_RELIABLE_TIMED: u8 = 0x02;
+const CHANNEL_TYPE_PARTIAL_RELIABLE_TIMED_UNORDERED: u8 = 0x82;
+
+const CHANNEL_TYPE_LEN: usize = 1;
+
+#[derive(Eq, PartialEq, Clone, Debug)]
+pub enum Error {
```

Comment:
please use thiserror crate for simplifying error handling.

and move all error enums to a dedicated lib/error.rs file.

Reply:
Done.